runtime/src/bank/tests.rs

```diff
@@ -1581,13 +1582,17 @@ impl Bank {
     }
 }
 
-#[test]
-fn test_rent_eager_collect_rent_in_partition() {
+#[test_case(true; "disable rent fees collection")]
+#[test_case(false; "enable rent fees collection")]
+fn test_rent_eager_collect_rent_in_partition(disable_rent_fees_collection: bool) {
     solana_logger::setup();
 
     let (mut genesis_config, _mint_keypair) = create_genesis_config(1_000_000);
     for feature_id in FeatureSet::default().inactive {
-        if feature_id != solana_sdk::feature_set::set_exempt_rent_epoch_max::id() {
+        if feature_id != solana_sdk::feature_set::set_exempt_rent_epoch_max::id()
+            && (!disable_rent_fees_collection
+                || feature_id != solana_sdk::feature_set::disable_rent_fees_collection::id())
+        {
```

Comment:
This won't work. Bank.deactivate_feature() only disables the feature for current bank. For future banks that passed epoch boundary, the feature will be re-activated since its account is in the genesis.